src/metrics/basic.py

```diff
@@ -0,0 +1,282 @@
+from itertools import groupby
+from numpy import append
+from label_data_utils import getGitHubData, getLabelData
+import datetime
+from easydict import EasyDict
+import math
+
+QueryConfig = {
+                'labelUnion': None,
+                'labelIntersect': None,
+                'repoIds': None,
+                'orgIds': None,
+                'repoNames': None,
+                'orgNames': None,
+                'userIds': None,
+                'userLogins': None,
+                'startYear': 2015,
+                'startMonth': 1,
+                'endYear': 2015,
+                'endMonth': 12,
+                'order': 'DESC',
+                'limit': 10,
+                'percision': 2,
+                'groupBy': None,
+                'groupTimeRange': None,
+                'options': None
+}
+
+def getMergedConfig(config):
+    defaultConfig = {
+        'startYear': 2015,
+        'startMonth': 1,
+        'endYear': datetime.datetime.today().year,
+        'endMonth': datetime.datetime.today().month,
+        'order': 'DESC',
+        'limit': 10,
+        'percision': 2,
+    } 
+    defaultConfig.update(config)
+    return defaultConfig
+
+
+def forEveryMonthByConfig(config, func):
+    return forEveryMonth(config.get('startYear'), config.get('startMonth'), config.get('endYear'), config.get('endMonth'), func)
+
+def forEveryMonth(startYear, startMonth, endYear, endMonth, func):
+    for y in range(startYear, endYear + 1):
+        begin_month = startMonth if y == startYear else 1
+        end_month = endMonth if y == endYear else 12
+        for m in range(begin_month, end_month + 1):
+            func(y, m)
+
+# Repo
+def getRepoWhereClauseForNeo4j(config):
+    def process(l):
+        data = getGitHubData([l])
+        data = EasyDict(data)
+        arr = []
+        if len(data.githubRepos) > 0: arr.append('r.id IN {}'.format(data.githubRepos))
```

Comment:
All `array` or `list` param format need no string construction manually, Python can convert list to string that database can understand in SQL. So remove all `[]` and string construction.